### PR TITLE
Fix SickRage/sickrage-issues/issues/341

### PR DESCRIPTION
**Pull request by fernandog**
https://github.com/SickRage/sickrage-issues/issues/341

@miigotu

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

File: SickBeard.py [hunk 1]
```diff
@@ -104,6 +104,15 @@ def __init__(self):
         self.consoleLogging = True
 
     @staticmethod
+    def clear_cache():
+        try:
+            cache_folder = ek(os.path.join, sickbeard.CACHE_DIR, 'mako') 
+            if os.path.isdir(cache_folder):
+                shutil.rmtree(cache_folder)
+        except Exception:
+            logger.log(u"Halt: Unable to remove the cache/mako directory!", logger.WARNING)
```
Comment (fernandog) [inline SickBeard.py]: @miigotu logger will work when called from line 308?

**Comment by neoatomic** [comment]
@fernandog I had the same issue of SR not booting. Tried your first commit and its working. (not tested the others yet)

**Comment by miigotu** [comment]
Lol guess my removing of cache/mako works a little too well lol. This should be ok,

File: SickBeard.py [hunk 2]
```diff
@@ -305,9 +316,8 @@ def start(self):
...
 
         # Clean up after update
-        if sickbeard.GIT_NEWVER:
-            shutil.rmtree(ek(os.path.join, sickbeard.CACHE_DIR, 'mako'))
-            sickbeard.GIT_NEWVER = False
+        if not self.clear_cache():
+            print u"Unable to remove the cache/mako directory!"
 
         if self.forcedPort:
             logger.log(u"Forcing web server to port " + str(self.forcedPort))
```
Comment (miigotu) [inline SickBeard.py]: does this logger line work?

**Comment by fernandog** [comment]
just fixup to remote the "halt" from the print...

File: SickBeard.py [hunk 3]
```diff
@@ -498,10 +508,9 @@ def shutdown(self, event):
                 except Exception:
                     pass
 
-            try:
-                logger.log(u"Halt: Trying to remove the cache/mako directory")
-                shutil.rmtree(ek(os.path.join, sickbeard.CACHE_DIR, 'mako'))
-            except Exception:
+            # Clean cache
+            logger.log(u"Halt: Trying to remove the cache/mako directory")
```
Comment (miigotu) [inline SickBeard.py]: probably dont need to log this one as info.

File: SickBeard.py (hunk 2 above)
Comment (fernandog) [inline SickBeard.py]: don't know. never used forced port and don't know how to force it

**Comment by fernandog** [comment]
@miigotu changed your comments

**Pull request merged**